### DISCUSSION TITLE
Do all the features that work with Java also work with Kotlin ?

### DISCUSSION THREAD

**Discussion by smaikk** [getgauge/gauge#1861] [General]
As we use Kotlin with Gauge-java, do all the features that work with Java also work with Kotlin, such as Parallel execution by using threads for example ?

**Comment by zabil** [comment]
It should work. Do you see any issues?

**Reply by smaikk** [reply]
I would like to suggest to my team to use gauge-java with java for an android application developed in kotlin, but my team prefers to use Kotlin for automation tests with appium.

So just to be sure that if I choose Kotlin I will have all the features offered by gauge-java.

**Comment by sriv** [comment]
I believe it won't be as seamless running kotlin tests as with java. Here are some shortcomings:

- gauge does not understand kotlin (yet) so it cannot compile or parse it.
- as a result, there needs to be a separate task that would compile kotlin and set GAUGE_CUSTOM_BUILD_PATH to the output dir.
- authoring experience will be mediocre since gauge cannot parse kotlin code.

So in summary, it can be made to work, but with some effort

**Reply by smaikk** [reply]
Thank you for your precise answer, it will help me to make the right choice.
I think that in the end, I will use Java so as not to take risks.
